Source/WebCore/dom/Document.cpp

```diff
@@ -11076,6 +11077,28 @@ void Document::securityOriginDidChange()
     m_permissionsPolicy = nullptr;
 }
 
+#if ENABLE(CONTENT_EXTENSIONS)
+
+ResourceMonitor* Document::resourceMonitor()
+{
+    return m_resourceMonitor.get();
+}
+
+void Document::setResourceMonitor(RefPtr<ResourceMonitor>&& resourceMonitor)
+{
+    m_resourceMonitor = WTFMove(resourceMonitor);
+}
+
+ResourceMonitor* Document::parentResourceMonitor()
+{
+    if (RefPtr parent = parentDocument())
+        return parent->resourceMonitor();
```

Comment:
Not totally sure I understand the implementation of this function. Is it supposed to return the first non-null resource monitor in the tree like this?

```
if (m_resourceMonitor)
  return m_resourceMonitor;
if (RefPtr parent = parentDocument())
  return parent->resourceMonitor();
return nullptr;
```

Otherwise, it seems like this method is equivalent to `topDocument().resourceMonitor()` since it walks the tree upwards until it hits the root document.